cg/meta/upload/fohm/fohm.py

```diff
@@ -147,79 +167,97 @@ def create_daily_delivery_folders(self) -> None:
+                name=report.sample_number
+            ).internal_id
+
+    def add_sample_internal_id_pangolin_report(self, reports: list[FohmPangolinReport]) -> None:
+        """Add sample internal id to Pangolin reports."""
+        for report in reports:
+            report.internal_id = self.status_db.get_sample_by_name(name=report.taxon).internal_id
+
+    def add_region_lab_to_reports(
+        self, reports: list[FohmComplementaryReport] | list[FohmPangolinReport]
+    ) -> None:
+        """Add region lab to reports."""
+        for report in reports:
+            report.region_lab = f"""{self.lims_api.get_sample_attribute(lims_id=report.internal_id, key="region_code").split(' ')[0]}"""
+            f"""_{self.lims_api.get_sample_attribute(lims_id=report.internal_id, key="lab_code").split(' ')[0]}"""
+
+    def link_sample_rawdata_files(
+        self, reports: list[FohmComplementaryReport] | list[FohmPangolinReport]
+    ) -> None:
+        """Hardlink samples raw data files to FOHM delivery folder."""
+        for report in reports:
+            for sample_id in report.internal_id:
+                sample: Sample = self.status_db.get_sample_by_internal_id(internal_id=sample_id)
+                bundle_name = sample.links[0].case.internal_id
+                version_obj: Version = self.housekeeper_api.last_version(bundle=bundle_name)
+                files = self.housekeeper_api.files(version=version_obj.id, tags=[sample_id]).all()
+                for file in files:
+                    if self._dry_run:
+                        LOG.info(
+                            f"Would have copied {file.full_path} to {Path(self.daily_rawdata_path)}"
+                        )
+                        continue
+                    shutil.copy(file.full_path, Path(self.daily_rawdata_path))
+
+    def create_pangolin_report_(self, reports: list[FohmPangolinReport]) -> None:
+        LOG.info("Creating aggregate Pangolin report")
+        unique_region_labs: set[str] = {report.region_lab for report in reports}
+        LOG.info(f"Regions in batch: {unique_region_labs}")
+        for region_lab in unique_region_labs:
             LOG.info(f"Aggregating data for {region_lab}")
-            pangolin_df = self.pangolin_dataframe[
-                self.aggregation_dataframe["region_lab"] == region_lab
+            region_lab_reports: list[FohmPangolinReport] = [
+                report for report in reports if report.region_lab == region_lab
             ]
             if self._dry_run:
-                LOG.info(pangolin_df)
+                LOG.info(region_lab_reports)
                 continue
-            pangolin_path = Path(
+            pangolin_report_file = Path(
                 self.daily_rawdata_path,
-                f"{region_lab}_{self.current_datestr}_pangolin_classification_format4.txt",
+                f"{region_lab}_{self.current_datestr}_pangolin_classification_format4{FileExtensions.TXT}",
```

Comment:
@Clinical-Genomics/prodbioinfo Is it possible to change file extension to CSV?

Reply:
I'm hesitant to rename this file as it's part of the ones we send to FOHM. Wouldn't be surprised if it breaks something on their end as they have it hard-coded somewhere. We could of course ask them if you see a huge benefit with renaming it

Reply:
Let's change it later if needed.